### ISSUE TITLE
[exporters/otlptracehttp] Got error messages despite export succeeded

### ISSUE THREAD

**Issue by songy23** [open-telemetry/opentelemetry-go#3706]
### Description

When using the `otlptracehttp` exporter to export trace to a custom endpoint, I got error messages despite export succeeded:
```
2023/02/10 10:34:31 traces export: failed to send to [my own trace agent]: 202 Accepted
```

/cc @dineshg13 

### Environment

- OS: MacOS
- Architecture: Apple M1
- Go Version: 1.19
- opentelemetry-go version: v1.13.0

### Steps To Reproduce

1. Have a mock backend that returns 202 upon successfully receiving OTLP HTTP trace payloads
2. Run `otlptracehttp` exporter against the mock backend
3. See error despite traces are successfully exported

### Expected behavior

No error logs upon `202 Accepted` responses for `otlptracehttp` exporter.

**Comment by MrAlias** [comment]
Please include a full run-able example of the error. Including the response your mock server is returning.

**Comment by songy23** [comment]
Hi @MrAlias I have a unit test to reproduce it and a fix in https://github.com/open-telemetry/opentelemetry-go/pull/3707. Please assign this issue to me.

**Comment by songy23** [comment]
This is currently blocked by the spec: https://github.com/open-telemetry/opentelemetry-specification/issues/3203

**Comment by pellared** [comment]
It looks like already addressed: https://github.com/open-telemetry/opentelemetry-go/blame/8d2c38d6057ff7b21a5f12ec65f4080943bf77ba/exporters/otlp/otlptrace/otlptracehttp/client.go#L169

Fixed in https://github.com/open-telemetry/opentelemetry-go/pull/4365

Please reopen or create a new issue if I am wrong.
